src/ruby/yast/profiler.rb

```diff
@@ -0,0 +1,71 @@
+
+require "yast"
+
+module Yast
+  class Profiler
+    class << self
+      include Yast::Logger
+
+      RESULT_PATH = "/var/log/YaST2/profiler_result.txt"
```

Comment:
NP: That target is writable only for *root*, if you have an Yast module which can run without *root* privileges than you cannot profile it.

My suggestions:

- Use a temporary file with `yast2-profile-` prefix, it ensures the file name is unique.
- Ask the user for the file name using [UI::AskForSaveFileName()](https://doc.opensuse.org/projects/YaST/SLES11/tdg/AskForSaveFileName.html)
- Pass the file name via environment, e.g. `Y2PROFILERFILE` (if not set then use one of the options above)